### PR TITLE
Feature/cnnwe 470

### PR DESCRIPTION
**Pull request by ghost**
@martytheparty @adslaton . Hi, I have made the changes so that the event listeners are now passed into the injector via the CNN.INJECTOR.registerEvents function and I bound the createDeferredForFeature to the CNN namespace so you could pass in a string to tell it to create a deferred object for a feature. Please merge if you both approve.

### CODE REVIEW

File: dist/injector.js [hunk 1]
```diff
@@ -94,30 +75,38 @@ window.NS.bundleHost = window.NS.bundleHost || '/';
...
-                NS.INJECTOR.loadFeature(resource);
-            });
-        };
-        /* onZonesAndDomReady is applicable when zones are not dynamically loaded */
-        jQuery(document).onZonesAndDomReady(loadFeature);
-        /* onZoneRendered is so the browser requests features faster when zone loading is on */
-        jQuery(document).onZoneRendered(loadFeature);
+    
+    loadFeature = function () {
+        jQuery('[data-cnn-resource]').each(function (idx, el) {
```
Comment (adslaton) [inline dist/injector.js:80]: This should be refactored to be `data-resource` or `data-bundle`.

Comment (ghost) [reply dist/injector.js:80]: If we do change, then a ticket has to be opened to change all the attributes that have been added in previous tickets for injector to be changed from data-cnn-resource to data-resouce or data-bundle. Do you still want to refactor it?

Comment (adslaton) [reply dist/injector.js:80]: Open a ticket to refactor. I am approving... Once Marty approves then I will merge and tag this change.

**Review by adslaton — APPROVED** [review]

**Review by martytheparty — APPROVED** [review]

File: dist/injector.js [hunk 2]
```diff
@@ -240,4 +229,4 @@ window.NS.bundleHost = window.NS.bundleHost || '/';
         }
         return promise;
     };
-})(window.NS, jQuery);
+})(window.FAI, jQuery);
```
Comment (adslaton) [inline dist/injector.js:232]: Change lines 13 and 14. Also change the README.md reference.

Comment (ghost) [reply dist/injector.js:232]: It has been done

File: dist/injector.js [hunk 3]
```diff
@@ -10,39 +10,20 @@
  *
  *
  */
-window.NS = window.NS || {};
-window.NS.bundleHost = window.NS.bundleHost || '/';
+window.FAI = window.NS || {};
```
Comment (adslaton) [inline dist/injector.js]: Almost... you need to set `window.FAI = window.FAI || {};`. You will need to update the reference in /example/index.html.

Comment (ghost) [reply dist/injector.js]: Thanks for the catch. I have updated it.

**Comment by adslaton** [comment]
Looks good!

**Pull request merged**